catalog/core/catalog-core-standardframework/src/main/java/ddf/catalog/history/Historian.java

```diff
@@ -164,14 +166,22 @@ public UpdateStorageResponse version(UpdateStorageRequest streamUpdateRequest,
         setSkipFlag(streamUpdateRequest);
         setSkipFlag(updateStorageResponse);
 
-        Collection<ReadStorageRequest> ids = getReadStorageRequests(updateStorageResponse);
+        List<Metacard> updatedMetacards = updateStorageResponse.getUpdatedContentItems()
+                .stream()
+                .filter(ci -> ci.getQualifier() == null || ci.getQualifier()
+                        .equals(""))
+                .map(ContentItem::getMetacard)
+                .filter(isVersionOrDeleted().negate())
+                .collect(Collectors.toList());
+
+        Collection<ReadStorageRequest> ids = getReadStorageRequests(updatedMetacards);
         if (ids.isEmpty()) {
             LOGGER.debug("No root content items to version");
             return updateStorageResponse;
         }
 
         Map<String, Metacard> metacards = query(forIds(fromStorageRequests(ids)));
-        Map<String, List<ContentItem>> oldContent = getOldContent(ids);
+        Map<String, List<ContentItem>> oldContent = getContent(ids);
```

Comment:
Renamed because it would fetch *any* content specified (by the `ReadStorageRequest`), it didn't discriminate against demographics 😄